runtime/lua/vim/treesitter.lua

```diff
@@ -404,7 +402,7 @@ end
 function M.start(bufnr, lang)
   bufnr = bufnr or a.nvim_get_current_buf()
   local parser = M.get_parser(bufnr, lang)
-  M.highlighter.new(parser)
+  M.highlighter.new(parser, { timeout = 1 })
```

Comment:
that might be a bit low for the tests... (any kind of timeout value is poison for CI, as the system load varies wildly :/)

Reply:
It's not really a timeout, it's more maximum time to occupy within each loop iteration. A parse is never cancelled because of this.